framework/CodeInterfaces/RAVEN/RAVENInterface.py

```diff
@@ -0,0 +1,257 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on Sept 10, 2017
+
+@author: alfoa
+"""
+from __future__ import division, print_function, unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default',DeprecationWarning)
+
+import os
+from  __builtin__ import any as b_any
+import copy
+import numpy as np
+from utils import utils
+from CodeInterfaceBaseClass import CodeInterfaceBase
+import DataObjects
+#import RavenData
+import csvUtilities
+
+class RAVEN(CodeInterfaceBase):
+  """
+    this class is used as part of a code dictionary to specialize Model.Code for RAVEN
+  """
+  def __init__(self):
+    CodeInterfaceBase.__init__(self)
+    self.printTag  = 'RAVEN INTERFACE'
+    self.outputPrefix = 'out~'
+    self.outStreamsNamesAndType = {} # Outstreams names and type {'outStreamName':[DataObjectName,DataObjectType]}
+    # pointer to the module that contains the function to modify and convert the sampled vars (optional)
+    # 2 methods are going to be inquired (if present and needed):
+    # - convertNotScalarSampledVariables
+    # - manipulateScalarSampledVariables
+    self.extModForVarsManipulation = None
+    # 'noscalar' = True if convertNotScalarSampledVariables exists in self.extModForVarsManipulation module
+    # 'scalar'   = True if manipulateScalarSampledVariables exists in self.extModForVarsManipulation module
+    self.hasMethods                = {'noscalar':False, 'scalar':False}
+    # inner workind directory
+    self.innerWorkingDir = ''
+    # linked DataObjects
+    self.linkedDataObjectOutStreamsNames = None
+
+  def addDefaultExtension(self):
+    """
+      This method sets a list of default extensions a specific code interface accepts for the input files.
+      This method should be overwritten if these are not acceptable defaults.
+      @ In, None
+      @ Out, None
+    """
+    self.addInputExtension(['xml'])
+
+  def _readMoreXML(self,xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class and initialize
+      some members based on inputs. This can be overloaded in specialize code interface in order to
+      read specific flags.
+      In this case, this is used for locating an external python module where the variables can be modified and the "vector" variables
+      can be splitted in multiple single variables
+      @ In, xmlNode, xml.etree.ElementTree.Element, Xml element node
+      @ Out, None.
+    """
+    if os.path.basename(xmlNode.find("executable").text) != 'raven_framework':
+      raise IOError(self.printTag+' ERROR: executable must be "raven_framework" (in whatever location)!')
```

Comment:
I decided to raise an error because in the raven_framework we do a lot of checks on the libraries that we do not do in Driver.py, so I thought it would be safer to force the user to use the "raven_framework" wrapper.

Do you think I should allow also to directly point to Driver.py?

Reply:
No, I would rather not allow pointing directly to Driver.py.  I think leaving it an error for now is fine, and if someone thinks of a good reason to use something else, we can change it at that point.